src/libs/kata-types/src/config/runtime_vendor.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! A sample for vendor to customize the runtime implementation.
+
+use super::*;
+use crate::{eother, sl};
+use slog::Level;
+/// Vendor customization runtime configuration.
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct RuntimeVendor {
+    /// Log level
+    #[serde(default)]
+    pub log_level: u32,
+
+    /// Prefix for log messages
+    #[serde(default)]
+    pub log_prefix: String,
+}
+
+impl ConfigOps for RuntimeVendor {
+    fn adjust_configuration(conf: &mut TomlConfig) -> Result<()> {
+        if conf.runtime.vendor.log_level > Level::Debug as u32 {
+            conf.runtime.debug = true;
+        }
+
+        Ok(())
+    }
+
+    /// Validate the configuration information.
+    fn validate(conf: &TomlConfig) -> Result<()> {
+        if conf.runtime.vendor.log_level > 10 {
+            warn!(
+                sl!(),
+                "log level {} in configuration file is invalid", conf.runtime.vendor.log_level
+            );
```

Comment:
Do we need the `warn` when returning an error?

Reply:
fixed